stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt

```diff
@@ -0,0 +1,129 @@
+                }
+            }
+
+            state<ClientState.UserState.UserAuthorized.Connected> {
+                onEvent<ChatClientEvent.DisconnectedEvent> { state, _ ->
+                    ClientState.UserState.UserAuthorized.Disconnected(state.connectionId, state.user, state.token)
+                }
+                onEvent<ChatClientEvent.DisconnectRequestedEvent> { _, _ -> ClientState.Idle }
+            }
+
+            state<ClientState.UserState.UserAuthorized.Disconnected> {
+                onEvent<ChatClientEvent.DisconnectedEvent> { _, _ -> stay() }
+                onEvent<ChatClientEvent.DisconnectRequestedEvent> { _, _ -> ClientState.Idle }
+                onEvent<ChatClientEvent.ConnectedEvent> { state, event ->
+                    ClientState.UserState.UserAuthorized.Connected(event.connectionId, event.user, state.token)
+                }
+            }
+
+            state<ClientState.AnonymousUserState.AnonymousUserPending.AnonymousPendingWithoutToken> {
+                onEvent<ChatClientEvent.TokenReceivedEvent> { _, event ->
+                    ClientState.AnonymousUserState.AnonymousUserPending.AnonymousPendingWithToken(event.token)
+                }
+            }
+
+            state<ClientState.AnonymousUserState.AnonymousUserPending.AnonymousPendingWithToken> {
+                onEvent<ChatClientEvent.ConnectedEvent> { state, event ->
+                    ClientState.AnonymousUserState.AnonymousUserAuthorized.AnonymousUserConnected(
+                        event.connectionId,
+                        event.user,
+                        state.token
+                    )
+                }
+            }
+
+            state<ClientState.AnonymousUserState.AnonymousUserAuthorized.AnonymousUserConnected> {
+                onEvent<ChatClientEvent.DisconnectedEvent> { state, _ ->
+                    ClientState.AnonymousUserState.AnonymousUserAuthorized.AnonymousUserDisconnected(
+                        state.connectionId,
+                        state.anonymousUser,
+                        state.token
+                    )
+                }
+                onEvent<ChatClientEvent.DisconnectRequestedEvent> { _, _ -> ClientState.Idle }
+            }
+
+            state<ClientState.AnonymousUserState.AnonymousUserAuthorized.AnonymousUserDisconnected> {
+                onEvent<ChatClientEvent.DisconnectedEvent> { _, _ -> stay() }
+                onEvent<ChatClientEvent.DisconnectRequestedEvent> { _, _ -> ClientState.Idle }
+                onEvent<ChatClientEvent.ConnectedEvent> { state, event ->
+                    ClientState.AnonymousUserState.AnonymousUserAuthorized.AnonymousUserConnected(
+                        event.connectionId,
+                        event.user,
+                        state.token
+                    )
+                }
+            }
+        }
+    }
+
+    internal val state = stateMachine.state
```

Comment:
This is a one-time assignment of this value, and not a getter that's executed every time - seems like a bug.